swaybar/render.c

```diff
@@ -286,6 +286,8 @@ static uint32_t render_status_block(struct render_context *ctx,
 		render_sharp_line(cairo, border_color, x_pos, y_pos,
 				block->border_left, render_height);
 		x_pos += block->border_left + margin;
+	} else if (block->urgent) {
+		x_pos += block->border_left + margin;
```

Comment:
What if `block->border` is transparent? Don't we still need to add the offset even if the block is not urgent?

I'd be completely fine with dropping the `border_color` "optimization". Pretty sure it doesn't make any meaningful difference.

Reply:
Correct, and this code does that: the only way to get a border without being urgent is to specify one on the block; `border_color` will reference `urgent_color` (if `urgent` set) or `border->color` otherwise.

It's not clear what value `border->color` has if the block has no specified border; I thought I covered all the permutations but I will verify this.